src/movies.js

```diff
@@ -1,22 +1,55 @@
 // Iteration 1: All directors? - Get the array of all directors.
 // _Bonus_: It seems some of the directors had directed multiple movies so they will pop up multiple times in the array of directors.
 // How could you "clean" a bit this array and make it unified (without duplicates)?
-function getAllDirectors(moviesArray) {}
+function getAllDirectors(moviesArray) {
+    return moviesArray.map(movies => movies.director)
+}
 
 // Iteration 2: Steven Spielberg. The best? - How many drama movies did STEVEN SPIELBERG direct?
-function howManyMovies(moviesArray) {}
+function howManyMovies(moviesArray) {
+    let dramaArr = moviesArray.filter((movie) => {
+        return movie.director === "Steven Spielberg" && movie.genre.includes("Drama")
+    })
+return dramaArr.length
+}
 
 // Iteration 3: All scores average - Get the average of all scores with 2 decimals
-function scoresAverage(moviesArray) {}
+function scoresAverage(moviesArray) {
+    if (!moviesArray.length) return 0
+    let sumScores = moviesArray.reduce((acc, movie) => {
+        if(movie.score) return acc + movie.score
+        else return acc
+    }, 0)
+    averageScore = sumScores / moviesArray.length
+    return Math.round(averageScore * 100) / 100
+}
 
 // Iteration 4: Drama movies - Get the average of Drama Movies
-function dramaMoviesScore(moviesArray) {}
+function dramaMoviesScore(moviesArray) {
```

Comment:
Instead of copying the logic of the scoresAverage function, you could've returned the scoresAverage function itself here. 
Like: `return scoresAverage(filterMovies);`